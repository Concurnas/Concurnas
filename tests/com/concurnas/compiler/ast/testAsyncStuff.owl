import System;

y = new Doer(56);
x int = func(69)!(new Doer(2));

{
	System.out.println("hi");//this is a jimport?
	System.out.println("there");
}!

{
	n int = 66;
}

m int =8;

(func&(? int,2))(3)!;

d=ax[x].sfun()(2)!;

sync{ ref2(3)!; }

