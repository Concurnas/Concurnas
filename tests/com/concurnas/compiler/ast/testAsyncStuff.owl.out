{
import System as System ;
y = new Doer ( 56 ) ;
x int = {
func ( 69 ) ;
}
! ( new Doer ( 2 ) ) ;
{
System . out . println ( " hi " ) ;
System . out . println ( " there " ) ;
}
! {
n int = 66 ;
}
m int = 8 ;
{
func & ( ? int , 2 ) ( 3 ) ;
}
! ;
d = {
ax [ x ] . sfun ( ) ( 2 ) ;
}
! ;
try {
com \. concurnas \. bootstrap \. runtime \. cps \. Fiber \. getCurrentFiber ( ) \. enterSync ( ) ;
{
ref2 ( 3 ) ;
}
! ;
}
finally {
com \. concurnas \. bootstrap \. runtime \. cps \. Fiber \. getCurrentFiber ( ) \. exitsync ( ) ;
}
}